windows/src/main/csharp/ch/cyberduck/ui/ViewModels/TransfersViewModel.cs

```diff
@@ -0,0 +1,535 @@
+            }
+        }
+    }
+
+    public ReadOnlyObservableCollection<TransferViewModel> SelectedTransfers => selectedTransfers;
+
+    public bool ToolbarCleanup
+    {
+        get => preferences.getBoolean("transfer.toolbar.cleanup");
+        set
+        {
+            OnPropertyChanging();
+            preferences.setProperty("transfer.toolbar.cleanup", value);
+            OnPropertyChanged();
+            OnPropertyChanged(nameof(ToolbarSeparatorLeft));
+        }
+    }
+
+    public bool ToolbarOpen
+    {
+        get => preferences.getBoolean("transfer.toolbar.open");
+        set
+        {
+            OnPropertyChanging();
+            preferences.setProperty("transfer.toolbar.open", value);
+            OnPropertyChanged();
+            OnPropertyChanged(nameof(ToolbarSeparatorRight));
+        }
+    }
+
+    public bool ToolbarReload
+    {
+        get => preferences.getBoolean("transfer.toolbar.reload");
+        set
+        {
+            OnPropertyChanging();
+            preferences.setProperty("transfer.toolbar.reload", value);
+            OnPropertyChanged();
+            OnPropertyChanged(nameof(ToolbarSeparatorLeft));
+        }
+    }
+
+    public bool ToolbarRemove
+    {
+        get => preferences.getBoolean("transfer.toolbar.remove");
+        set
+        {
+            OnPropertyChanging();
+            preferences.setProperty("transfer.toolbar.remove", value);
+            OnPropertyChanged();
+            OnPropertyChanged(nameof(ToolbarSeparatorLeft));
+        }
+    }
+
+    public bool ToolbarResume
+    {
+        get => preferences.getBoolean("transfer.toolbar.resume");
+        set
+        {
+            OnPropertyChanging();
+            preferences.setProperty("transfer.toolbar.resume", value);
+            OnPropertyChanged();
+            OnPropertyChanged(nameof(ToolbarSeparatorLeft));
+        }
+    }
+
+    public bool ToolbarShow
+    {
+        get => preferences.getBoolean("transfer.toolbar.show");
+        set
+        {
+            OnPropertyChanging();
+            preferences.setProperty("transfer.toolbar.show", value);
+            OnPropertyChanged();
+            OnPropertyChanged(nameof(ToolbarSeparatorRight));
+        }
+    }
+
+    public bool ToolbarStop
+    {
+        get => preferences.getBoolean("transfer.toolbar.stop");
+        set
+        {
+            OnPropertyChanging();
+            preferences.setProperty("transfer.toolbar.stop", value);
+            OnPropertyChanged();
+            OnPropertyChanged(nameof(ToolbarSeparatorLeft));
+        }
+    }
+
+    public bool ToolbarTrash
+    {
+        get => preferences.getBoolean("transfer.toolbar.trash");
+        set
+        {
+            OnPropertyChanging();
+            preferences.setProperty("transfer.toolbar.trash", value);
+            OnPropertyChanged();
+            OnPropertyChanged(nameof(ToolbarSeparatorLeft));
+        }
+    }
+
+    public bool ToolbarSeparatorLeft
```

Comment:
Ugly, but extracting a TransferToolbarButtonProvider which dynamically changes the toolbar in the Transfers window would be more complicated.